### ISSUE TITLE
TestAdapter requires constant strings for test names

### ISSUE THREAD

**Issue by mryandot** [adamdriscoll/poshtools#456]
I was having an issue where the Test Explorer was not finding my tests.  After some experimenting, it appears that it's because I was using interpolated strings for my Describe statements.  The [Pester course on Pluralsight](https://app.pluralsight.com/library/courses/powershell-testing-pester) uses variables in test names fairly extensively, and I couldn't find any other documentation that discourages that.

I would guess the issue is that [GetFunctionName in PowerShellTestDiscoverer.cs](https://github.com/adamdriscoll/poshtools/blob/9270fcc516bd9a5831c8edfbc66afb51590d0282/PowerShellTools.TestAdapter/PowerShellTestDiscoverer.cs#L126-L153) returns the empty string if the -Name parameter is not a StringConstantExpressionAst (mine were ExpandableStringExpressionAst). While I haven't set up a full debugging environment for poshtools, Reflector shows the TestCase constructor throws an exception on an empty name.  My brief testing on a file with multiple describe statements shows none of the tests are registered if any of them have an invalid name, which would support my assumptions.

It looks like poshtools can't practically support many practices suggested in that course, but it would be nice if it could at least let the user know about the issue; an entry in the Error Log (info or warning), and/or writing a message to the Output window (where the Wiki suggests we look) would help.

**Comment by adamdriscoll** [comment]
Good point. I think it'd be really hard to determine the value of the variable since it's just doing static code analysis to figure out those names but an error or warning might work. I'm very open to suggestions though.

**Comment by mryandot** [comment]
I don't think there's a better fix than notification on the poshtools side.  On the developer side, a few thoughts occur:

*  There's some TODO code in the discovery logic that might change this, but for now the Test Explorer only seems to look at the Describe blocks.  Don't use fancy code for describe blocks; save it for Contexts and It blocks.
* Most of the variables and loops used in the scripts presented in the course were used for generating boilerplate tests, so it might be handy to have support for Design-Time T4 templates in PowerShell projects (.pssproj).
* Use a different file name so you know that Test Explorer will ignore them.  The kinds of boilerplate tests the trainer was using could probably be left for a build script or similar tool; they wouldn't necessarily need to be integrated in Visual Studio.
